### DISCUSSION TITLE
stellarium on the Purism L5 mobile phone

### DISCUSSION THREAD

**Discussion by gurucubano** [Stellarium/stellarium#2831] [Q&A]
Hello,
I own a Linux phone (the Purism L5) which is Debian based. stellarium is not working on this, it initializes, reads in data, objects, but does not present the scene. A full log.txt of such a start is here:
https://forums.puri.sm/t/stellarium-app-on-byzantium/15123/2
Is there any chance at all to make this work? Thanks in advance.

**Comment by 10110111** [comment]
Might be able to work with `LIBGL_ALWAYS_SOFTWARE=1` environment variable. But the following message makes me think it's worse:
```
qt.qpa.xcb: xcb_shm_create_segment() can't be called for size 17179344900
```
Looks like Qt wants a 16 GiB shared memory segment, which is... suspicious.

**Reply by gurucubano** [reply]
Thanks. I've set this env var. The Qt message about the oversized remains.

**Reply by 10110111** [reply]
Something funny is happening inside Qt. I suppose we get `tlwRect.size()` with enormous values in `QWidgetRepaintManager::paintAndFlush()`. (Looks like "tlw" means "top-level window" from what I can find in `qwidgetrepaintmanager.cpp`.)
Someone will have to run a debugging session (with a debug build of Qt) on a device like yours to see what's going on.

**Comment by gurucubano** [comment]
Last night I compiled 1.1 from source, directly om the L5 (took ~4h). Compilation went fine, but the test start directly from `/home/purism/stellarium-1.1/build/unix/src/stellarium` shows the same issue:
```
...
Loaded plugin "Satellites"
[Satellites] loading catalogue file: "/home/purism/.stellarium/modules/Satellites/satellites.json"
Loaded plugin "SolarSystemEditor"
Using the ssystem_minor.ini file that already exists in the user directory...
QOpenGLFramebufferObject: Framebuffer incomplete attachment.
QOpenGLFramebufferObject: Framebuffer incomplete attachment.
QOpenGLFramebufferObject: Framebuffer incomplete attachment.
QOpenGLFramebufferObject: Framebuffer incomplete attachment.
QOpenGLFramebufferObject: Framebuffer incomplete, missing attachment.
qt.qpa.xcb: xcb_shm_create_segment() can't be called for size 17179344900, maximumallowed size is 4294967295
QWidget::paintEngine: Should no longer be called
...
```

**Comment by gurucubano** [comment]
One can google for the string `xcb_shm_create_segment() can't be called for size 17179344900` and there are a few other such cases with other pieces of software, some pointing to screen size 4k or V4L issues. The screen of the device in question, the L5, is 720x1440. Maybe this is not correctly detected by stellarium?

**Reply by 10110111** [reply]
All detection stellarium does is via Qt. And here it's not even stellarium's action, it's Qt internals. This is why I said one needs a debugging session to look into Qt, this is why you need a debug build of Qt for this.

**Comment by gurucubano** [comment]
I followed a hint in the L5 forum:
https://forums.puri.sm/t/stellarium-app-on-byzantium/15123/17

The env var was not set in general. If I set it in the small shell script which should bring up stellarium, this spills out on STDOUT or STDERR:
```
purism@pureos:~$ cat /tmp/stellarium.stdout
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
qt.qpa.plugin: Could not find the Qt platform plugin "wayland" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.

Aborted
```
Are those plugins code of stellarium or of our L5 installation?

**Reply by 10110111** [reply]
These are Qt plugins. Looks like Qt has been built without Wayland support (but with X11).
